core/page_store.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+
+#include "base/log.hpp"
+#include "core/page.hpp"
+
+namespace husky {
+
+typedef std::unordered_map<size_t, Page*> PageMap;
+
+typedef std::unordered_set<Page*> PageSet;
+
+class PageStore {
+   public:
+    static Page* create_page() {
+        PageMap& page_map = get_page_map();
+        Page* page;
+        if (page_set_->size() != 0) {
```

Comment:
No idea about this check. Can you explain what `PageMap` and `PageSet` are used for?